src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -551,8 +933,13 @@ class module_sram : public module_impl
     // create bo combined size of all ctrlcodes
     size_t sz = std::accumulate(data.begin(), data.end(), static_cast<size_t>(0), [](auto acc, const auto& ctrlcode) {
       return acc + ctrlcode.size();
-    });
-    m_buffer = xrt::bo{m_hwctx, sz, xrt::bo::flags::cacheable, 1 /* fix me */};
+      });
+    if (sz == 0) {
+      std::cout << "instruction buf is empty" << std::endl;
+      return;
+    }
+
+    m_buffer = xrt::bo{ m_hwctx, sz, xrt::bo::flags::cacheable, 1 /* fix me */ };
```

Comment:
What needs to be fixed in this line?

Reply:
The API requiring `1` for memory bank is awkward.   The `xrt:;bo::cacheable` flag is a bad hack also.